### PR TITLE
Use trans_arg_datum in trans_args_under_call_abi

### PR DESCRIPTION
**Pull request by dotdash**
The logic for the argument translation was duplicated here.

### CODE REVIEW

File: src/librustc_trans/trans/callee.rs [hunk 1]
```diff
@@ -934,12 +934,13 @@ fn trans_args_under_call_abi<'blk, 'tcx>(
...
                     });
-                let arg_datum = arg_datum.to_expr_datum();
-                let arg_datum =
-                    unpack_datum!(bcx, arg_datum.to_rvalue_datum(bcx, "arg"));
-                let arg_datum =
-                    unpack_datum!(bcx, arg_datum.to_appropriate_datum(bcx));
-                llargs.push(arg_datum.add_clean(bcx.fcx, arg_cleanup_scope));
+                llargs.push(unpack_result!(bcx, {
+                    trans_arg_datum(bcx,
+                                    field_types[i],
```
Comment (eddyb) [inline src/librustc_trans/trans/callee.rs]: Couldn't this whole `for` loop be written as `llargs.extend(field_types.map(|field_ty| ...))`?

Comment (dotdash) [inline src/librustc_trans/trans/callee.rs]: You need index in the call to `adt::trans_field_ptr`, doesn't seem worthwhile to replace the loop with something that then has to recreate the counter.

Comment (dotdash) [inline src/librustc_trans/trans/callee.rs]: Done

**Pull request merged**